src/gradient.class.js

```diff
@@ -95,18 +95,59 @@
      */
     offsetY: 0,
 
+    /**
+     * A transform matrix to apply to the gradient before paiting.
+     * It shares the same name with the svg property but behaves differently.
+     * @type Array[Number]
+     * @default null
+     */
+    gradientTransform: null,
+
+    /**
+     * coordinates units for coords.
+     * If `pixels`, the number of cords are in the same unit of width/ height.
+     * If set as `percentage` the coords are still a number, but 1 means 100% of width
+     * for the X and 100% of the height for the y. It can be bigger than 1 and negative.
+     * @type String pixels || percentage
+     * @default 'pixels'
+     */
+    gradientUnits: 'pixels',
+
+    /**
+     * Gradient type
+     * @type String linear || radial
+     * @default 'pixels'
+     */
+    type: 'linear',
+
     /**
      * Constructor
-     * @param {Object} [options] Options object with type, coords, gradientUnits and colorStops
+     * @param {Object} options Options object with type, coords, gradientUnits and colorStops
+     * @param {Object} [options.type] gradient type
+     * @param {Object} [options.gradientUnits] gradient units
+     * @param {Object} [options.offsetX] SVG import compatibility
+     * @param {Object} [options.offsetY] SVG import compatibility
+     * @param {Object} options.coords contains the coords of the gradient
+     * @param {Array[Object]} options.colorStops contains the colorstops.
+     * @param {Number} [options.coords.x1]
+     * @param {Number} [options.coords.y1]
+     * @param {Number} [options.coords.x2]
+     * @param {Number} [options.coords.y2]
+     * @param {Number} [options.coords.r1]
+     * @param {Number} [options.coords.r2]
```

Comment:
options.coords should probably be next to options.coords.x1.  Move colorstops up one line?